src/Plugin.php

```diff
@@ -95,6 +95,19 @@ public function setDeprecationHandler($service)
             $previousHandler = set_error_handler(
                 function ($code, $message, $file, $line, $context = null) use (&$previousHandler) {
                     if ($code == E_USER_DEPRECATED || $code == E_DEPRECATED) {
+                        // In PHP 8.0+ the $context variable has been removed from the set_error_handler callback
+                        // Therefore we need to fetch the correct file and line string ourselves
+                        if (PHP_VERSION_ID >= 80000) {
+                            $trace = debug_backtrace();
+                            foreach ($trace as $idx => $traceEntry) {
+                                if ($traceEntry['function'] !== 'trigger_error') {
+                                    continue;
+                                }
+                                $file = $trace[$idx + 2]['file'];
+                                $line = $trace[$idx + 2]['line'];
```

Comment:
This would be a good place to apply the stack offset defined in the deprecation warning but getting that data here is not straightforward.

Overall this is still an improvement from what we have currently and is now wrong in fewer scenarios. While we can't entirely close out the linked issue we can make incremental progress that is a solution for a large portion of scenarios.